src/key/keyExchange.ts

```diff
@@ -0,0 +1,38 @@
+import { toArray, toHex } from "../utility";
+
+/**
+ * @hidden
+ */
+const nacl = require("tweetnacl");
+
+export type X25519Public = string;
+export type X25519Private = string;
+
+/**
+ * Gets an ECDH session key for encryption and decryption between two parties
+ * @param otherPublicStr 32 byte hexadecimal string of the other side public key
+ * @param myPrivateStr 32 byte hexadecimal string of my side private key
+ * @returns 32 byte hexadecimal string of the shared secret
+ */
+export const exchange = (
+    otherPublicStr: X25519Public,
+    myPrivateStr: X25519Private
+): string => {
+    const otherPublic = toArray(otherPublicStr);
+    const myPrivate = toArray(myPrivateStr);
+    const q = nacl.scalarMult(myPrivate, otherPublic);
```

Comment:
How about renaming the `q` to `shared_secret` or other names?
And `otherPublicStr` to `otherPublic`, `myPrivateStr` to `myPrivate` too.

Reply:
Ok, I just use commonly used variable name `q` for ECDH result but `sharedSecret` is better in this higher level.
I love variable name shadowing but it is impossible in `Javascript`.
I'll rename `otherPublicStr` to `otherPublic` and the previous `otherPublic` to `GroupElement` which fits well with the 
 argument name of `scalarMult`